nwb_conversion_tools/datainterfaces/behavior/movie/movie_utils.py

```diff
@@ -141,3 +139,64 @@ def __exit__(self, *args):
 
     def __del__(self):
         self.vc.release()
+
+
+class MovieDataChunkIterator(GenericDataChunkIterator):
+    """DataChunkIterator specifically for use on RecordingExtractor objects."""
+
+    def __init__(
+        self,
+        movie_file: FilePathType,
+        buffer_gb: float = None,
+        chunk_shape: tuple = None,
+        stub_test: bool = False,
+    ):
+        self.video_capture_ob = VideoCaptureContext(movie_file)
+        self._full_frame_size_mb, self._full_frame_shape = self._get_frame_details()
+        if stub_test:
+            self.video_capture_ob.frame_count = 10
+        super().__init__(
+            buffer_gb=buffer_gb,
+            chunk_shape=chunk_shape,
+            display_progress=True,
+        )
+
+    def _get_default_chunk_shape(self, chunk_mb):
+        """Shape is either one frame or a subset: scaled frame size but with all pixel colors"""
+        num_frames = int(chunk_mb // self._full_frame_size_mb)
+        num_frames = 1 if num_frames is 0 else num_frames
+        return (num_frames,) + self._full_frame_shape[1:]
+
+    def _get_default_buffer_shape(self, buffer_gb):
+        """Buffer shape is a multiple of frame shape along the frame dimension."""
+        assert buffer_gb >= self._full_frame_size_mb / 1e3, f"provide buffer size >= {self._full_frame_size_mb/1e3} GB"
+        frames_count = self._get_default_shape_movie(
+            buffer_gb * 1e3, self._full_frame_shape[:1], self._full_frame_size_mb, self._maxshape[:1]
+        )
+        return tuple([frames_count[0], *self._maxshape[1:]])
```

Comment:
Why is the inner list wrapping needed here? I like what you did above with `(num_frames,) + self._full_frame_shape[1:]`

Reply:
because `self._full_frame_shape` is a tuple (which I define in this code) but `self._maxshape` is defined in the parent class elsewhere which we have no control over. I dont want to assume it to be a tuple. 
I can do: `frames_count + tuple(self._maxshape[1:])` instead